src/autodiff_model.jl

```diff
@@ -51,26 +51,23 @@ end
 
 show_header(io :: IO, nlp :: ADNLPModel) = println(io, "ADNLPModel - Model with automatic differentiation")
 
-function ADNLPModel(f, x0::AbstractVector; y0::AbstractVector = eltype(x0)[],
-                    lvar::AbstractVector = eltype(x0)[], uvar::AbstractVector = eltype(x0)[],
+function ADNLPModel(f, x0::AbstractVector;
+                    lvar::AbstractVector = fill(-Inf, length(x0)), uvar::AbstractVector = fill(Inf, length(x0)),
                     lcon::AbstractVector = eltype(x0)[], ucon::AbstractVector = eltype(x0)[],
-                    c = (args...)->throw(NotImplementedError("cons")),
+                    c = x -> eltype(x0)[], y0::AbstractVector = eltype(x0)[],
                     name::String = "Generic", lin::AbstractVector{<: Integer}=Int[])
```

Comment:
This looks fine but I find there's a discrepancy between the way we treat the bounds (without explicit AbstractVectors) and the constraints. What if there were another ADNLPModel constructor that also takes `ncon` and `c` as positional parameters, and where `lcon`, `ucon` and `y0` have default values defined with `fill!(-Inf, ncon)`, etc.?

Reply:
I'm creating constructors for each traditional situation:
- `ADNLPModel(f, x0, lvar, uvar)`
- `ADNLPModel(f, x0, ncon, ceq)`
- `ADNLPModel(f, x0, cineq, lcon, ucon)`
- `ADNLPModel(f, x0, lvar, uvar, ncon, ceq)`
- `ADNLPModel(f, x0, lvar, uvar, cineq, lcon, ucon)`
what do you think?

Reply:
In a different PR

Reply:
What does `ADNLPModel(f, x0, ncon, ceq)` do? Construct a zero rhs?

It may be enough to write
* `ADNLPModel(f, x0, lvar, uvar)`
* `ADNLPModel(f, x0, lvar, uvar, c, lcon, ucon)`

no?

Reply:
Yes, zero rhs. I think we can reduce to
- `ADNLPModel(f, x0, lvar, uvar)`
- `ADNLPModel(f, x0, c, lcon, ucon)`
- `ADNLPModel(f, x0, lvar, uvar, c, lcon, ucon)`

Reply:
Ok!